### PR TITLE
correct offset by read ahead bytes when using SEEK_CUR/use ID3v2Tag instead of tag 

### PR DESCRIPTION
**Pull request by pbhd**
for underlying file even when seekoff is called with relative offset,
because the underlying file is used to do buffered reads and is therefore at another
position (Usually the buffer size is around 0x10000).
This fix might also help with bug #16158
There is written more in http://forum.kodi.tv/showthread.php?tid=239373, without the fix chunks after the data-chunk in riff-files cannot be found...

### CODE REVIEW

**Comment by pbhd** [comment]
The 2nd commit makes the now correctly read id3v2-tags available to the Method CTagLoaderTagLib::Load

File: xbmc/filesystem/File.cpp [hunk 1]
```diff
@@ -1080,7 +1080,7 @@ CFileStreamBuffer::pos_type CFileStreamBuffer::seekoff(
 
   int64_t position = -1;
   if(way == std::ios_base::cur)
-    position = m_file->Seek(offset, SEEK_CUR);
+    position = m_file->Seek(pos+offset, SEEK_SET); // Dont seek from current pos, because m_file does readahead and so has wrong position
```
Comment (FernetMenta) [inline xbmc/filesystem/File.cpp]: why don't you correct offset by readahead bytes? there may be a reason why clients call this with SEEK_CUR

Comment (pbhd) [inline xbmc/filesystem/File.cpp]: Good idea, will try it out...

File: xbmc/filesystem/File.cpp [hunk 2]
```diff
@@ -1080,7 +1081,7 @@ CFileStreamBuffer::pos_type CFileStreamBuffer::seekoff(
 
   int64_t position = -1;
   if(way == std::ios_base::cur)
-    position = m_file->Seek(offset, SEEK_CUR);
+    position = m_file->Seek(offset-aheadbytes, SEEK_CUR);
```
Comment (Razzeee) [inline xbmc/filesystem/File.cpp]: You missed to remove a space here, when you reworked it.

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by fritsch** [comment]
Please squash the first with the 3rd and 4th commit, please - _after_ jenkins build is done. Thanks much in advance.

**Comment by pbhd** [comment]
Hope I did the squash the right way :)

**Comment by fritsch** [comment]
Sadly not - let's try again :-)

1) backup your branch
git branch shouse

2) fetch origin, which we assume is kodi upstream - if not change the to be fetched name
git fetch origin

3) rebase your changes onto our master
Here you can pick and squash patches, e.g. change pick with squash and the the top patch will be squashed with the one below it.

git rebase -i origin/master

4) force push in your own master
git push -f pbhd master

Assumptions:
origin: kodi repo
sportica: your local remote for your repo

Btw. you should use feature branches for developing

**Comment by pbhd** [comment]
Hmm, was quite stuck.... (I'm a git git)
If it's ok now, then the important thing is the -f with git push (and not doing a pull before the push as git recomends when not using -f
It it's not ok now, then I'm still stuck... Still sweating

**Comment by fritsch** [comment]
Yeah looks good. 

If you want to rebase again, you could reword the commits, please. In that case choose reword instead of squash. Use something like: 

File: Do not seek outside of buffer area
and
TagLoaderTagLib: Use ID3v2Tag loader

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by pbhd** [comment]
Build failed with taglib 1.8, so added a version check 4b3a6a5

**Comment by stefansaraev** [comment]
jenkins build this please

**Comment by stefansaraev** [comment]
ups. taglib version check is wrong 
```
#if (TAGLIB_MAJOR_VERSION == 1 && TAGLIB_MINOR_VERSION < 9)
id3v2 = wavFile->tag();
#else
id3v2 = wavFile->ID3v2Tag();
#endif
```
^ this would make more sense.

**Comment by fritsch** [comment]
That would break for 0.8 :-) See my comment in code.

**Comment by pbhd** [comment]
There seems to be a problem with two commits in one pull-request? From the [Build-output](http://jenkins.kodi.tv/job/Android-ARM/5658/console) it looks like a previous version of my modified TagLoaderTagLib.cpp
was used, because the Error-Line in the Build-output is 221
and in the latest commit 4b3a6a5 of the file the corresponding method-call is on line 222

**Comment by fritsch** [comment]
It was just aborted ...

As there never was a 0.8/0.9 libtag version your patch is good as is, via: https://taglib.github.io/

jenkins build this please

**Comment by mkortstiege** [comment]
@fritsch is this one good to go?

**Comment by stefansaraev** [comment]
yes @mkortstiege

**Comment by mkortstiege** [comment]
jenkins build and merge.

**Pull request merged**